### PR TITLE
Fix phone number formatting. Add links.

### PR DESCRIPTION
**Pull request by mpelzsherman**
## Description
Add links to all facility phone numbers on both result and detail pages.

## Testing done
Manual

## Screenshots
Kalispell_VA_Clinic___Veterans_Affairs

## Acceptance criteria
- [x] Unique phone numbers are displayed only once.
- [x] Any phone numbers (main and/or mental health) displayed on search result cards will have the visual appearance and functionality of a link.
- [x] Any phone numbers (main and/or mental health) displayed on facility detail pages will have the visual appearance and functionality of a link.
- [x] Any phone numbers (main and/or mental health) displayed within Facility Locator will be accessible to users with assistive technology.
- [x]  A Mental health phone number and a Main phone number is displayed whenever both are present in the data (even if the phone number itself is the same).
- [x] Axe checks for this functionality pass
- [x] Zoom testing passes
- [x]  Keyboard checks pass
- [x]  Screenreader checks pass

## Definition of done
- [x] Events are logged appropriately
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/sass/facility-locator.scss [hunk 1]
```diff
@@ -650,6 +650,10 @@ $facility-locator-color-gray: #ccc;
   text-decoration: none;
 }
 
+.facility-phone-link {
```
Comment (brangi) [inline src/applications/facility-locator/sass/facility-locator.scss:653]: Can we use values from https://design.va.gov/utilities/margins#margin-left instead?

File: src/applications/facility-locator/components/search-results/LocationPhoneLink.jsx [hunk 2]
```diff
@@ -69,14 +54,8 @@ const LocationPhoneLink = ({ location, from, query }) => {
...
-      {renderPhoneNumber('Main Number', null, phone.main, 'phone', null, from)}
-      {renderPhoneNumber(
-        'Mental Health',
-        null,
-        phone.mentalHealthClinic,
-        null,
-        from,
-      )}
+      {renderPhoneNumber('Main Number', null, phone.main, from)}
+      {renderPhoneNumber('Mental Health', null, phone.mentalHealthClinic, from)}
```
Comment (brangi) [inline src/applications/facility-locator/components/search-results/LocationPhoneLink.jsx:58]: `from` might have an issue by not being passed correctly but I will confirm

**Comment by mpelzsherman** [comment]
Working on using the [Telephone](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design/components/telephone/) component.

File: src/applications/facility-locator/components/search-results/LocationPhoneLink.jsx [hunk 3]
```diff
@@ -1,40 +1,32 @@
...
   if (!phone) {
     return null;
   }
 
   const re = /^(\d{3})[ -]?(\d{3})[ -]?(\d{4})[ ]?(x?)[ ]?(\d*)/;
+  const formattedPhoneNumber = phone
+    .replace(re, '$1-$2-$3 $4$5')
+    .replace(/x$/, '');
+  const extension = phone.replace(re, '$4$5').replace(/x$/, '');
+  const contact = phone.replace(re, '$1$2$3');
```
Comment (ncksllvn) [inline src/applications/facility-locator/components/search-results/LocationPhoneLink.jsx:16]: @Mottie fysa example use of the Telephone component

File: src/applications/facility-locator/components/search-results/LocationPhoneLink.jsx (hunk 2 above)
Comment (mpelzsherman) [reply src/applications/facility-locator/components/search-results/LocationPhoneLink.jsx:62]: Sorry @brangi - I don't follow. Please explain?

**Comment by brangi** [comment]
Looking good! but this could use a space in-between `at6` maybe?
Screen Shot 2020-07-10 at 5 48 26 PM

**Review by brangi — APPROVED** [review]

**Review by bkjohnson — APPROVED** [review]

**Comment by mpelzsherman** [comment]
@brangi @bkjohnson - I did some refactoring and added some unit tests. Can you please re-review?

**Comment by mpelzsherman** [comment]
> could use a space in-between `at6` maybe?

Fixed this, @brangi. Good catch!

**Review by brangi — APPROVED** [review]
LGTM

**Pull request merged**